Framework/API/inc/MantidAPI/AlgorithmFactory.h

```diff
@@ -108,6 +114,9 @@ class MANTID_API_DLL AlgorithmFactoryImpl final
   const std::vector<std::string> getKeys() const override;
   const std::vector<std::string> getKeys(bool includeHidden) const;
 
+  /// Get an algorithms name from the alias map
+  std::string getNameFromAliasMap(const std::string &algorithmName) const;
```

Comment:
This is a good addition so that the content can queried and it allows callers to simplify their code. May I suggest a slightly different name, something like `getRealNameFromAlias`? The `map` part is related to the implementation and we don't want any calling code to know that as we've encapsulated it away. Also maybe the argument could be called just `alias`? It's then a little clearer what each part means as name is a bit generic.

I also think this might be a good use for [`boost::optional`](https://www.boost.org/doc/libs/1_61_0/libs/optional/doc/html/boost_optional/quick_start.html#boost_optional.quick_start.optional_return_values). In this case there might be an alias or not but neither case warrants an exception as you've already spotted and used an empty string to indicate no alias. `boost::optional` is a nice way of encapsulate that some operation might return a value or none but it allows you distinguish this from just assuming empty string means not found. In other cases empty string might be valid and then you can't use that as a flag value.

All in all I think our method can look like:
```c++
boost::optional<std::string> getRealNameFromAlias(const std::string &alias) const noexcept;
```

Does that make sense?

Reply:
Thanks for telling me about this! I've never seen this before